indexing-service/src/main/java/org/apache/druid/indexing/common/task/AbstractTask.java

```diff
@@ -186,7 +187,7 @@ public final TaskStatus run(TaskToolbox taskToolbox) throws Exception
   public abstract TaskStatus runTask(TaskToolbox taskToolbox) throws Exception;
 
   @Override
-  public void cleanUp(TaskToolbox toolbox, TaskStatus taskStatus) throws Exception
+  public void cleanUp(TaskToolbox toolbox, @Nullable TaskStatus taskStatus) throws Exception
```

Comment:
Although no harm to annotate nullable here, I think taskStatus won't be possible null as it will be set to a value in all the code path.